src/aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -582,12 +582,16 @@ HttpResponseOutcome AWSClient::AttemptOneRequest(const std::shared_ptr<Aws::Http
         }
     }
 
-    if (DoesResponseGenerateError(httpResponse) || request.HasEmbeddedError(httpResponse->GetResponseBody(), httpResponse->GetHeaders()))
+    if (DoesResponseGenerateError(httpResponse) )
     {
         AWS_LOGSTREAM_DEBUG(AWS_CLIENT_LOG_TAG, "Request returned error. Attempting to generate appropriate error codes from response");
         auto error = BuildAWSError(httpResponse);
         return HttpResponseOutcome(std::move(error));
     }
+    else if(request.HasEmbeddedError(httpResponse->GetResponseBody(), httpResponse->GetHeaders()))
+    {
+        return HttpResponseOutcome(std::move(BuildAWSErrorFromResponseBody(httpResponse)) );
```

Comment:
gives a clang warning `error: moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]`

should update this to use a initializer list

```cpp
return {BuildAWSErrorFromResponseBody(httpResponse)};
```

Reply:
ack